syncstorage-postgres/src/db/batch_impl.rs

```diff
@@ -12,42 +27,234 @@ impl BatchDb for PgDb {
     async fn create_batch(
         &mut self,
         params: params::CreateBatch,
-    ) -> Result<results::CreateBatch, Self::Error> {
-        todo!()
+    ) -> DbResult<results::CreateBatch> {
+        let batch_id = Uuid::new_v4();
+        let user_id = params.user_id.legacy_id as i64;
+        let collection_id = self.get_or_create_collection_id(&params.collection).await?;
+        let timestamp_ms = self.timestamp().as_i64();
+        let expiry = timestamp_to_naive_datetime(timestamp_ms + BATCH_LIFETIME)?;
```

Comment:
I've added `SyncTimestamp::as_naive_datetime` in https://github.com/mozilla-services/syncstorage-rs/pull/1912, when that lands I'd say it'd be more expicit/preferable to shift time via chrono
```suggestion
        let expiry = self.timestamp().as_naive_datetime() + chrono::TimeDelta::milliseconds(BATCH_LIFETIME)?;
```